spec/unit/provider/service/smf_spec.rb

```diff
@@ -174,17 +174,42 @@
   end
 
   describe "when restarting" do
-    it "should call 'svcadm restart /system/myservice'" do
-      @provider.expects(:texecute).with(:restart, ["/usr/sbin/svcadm", :restart, "/system/myservice"], true)
-      @provider.expects(:wait).with('online')
-      @provider.restart
-    end
 
     it "should error if timeout occurs while restarting the service" do
-      @provider.expects(:texecute).with(:restart, ["/usr/sbin/svcadm", :restart, "/system/myservice"], true)
+      @provider.expects(:texecute).with(:restart, ["/usr/sbin/svcadm", :restart, "-s", "/system/myservice"], true)
```

Comment:
@shawnferry the spec failure is happening here because `Facter.value(:kernelrelease)` isn't being stubbed in the test, so Puppet is falling back to the restart `else` clause where we don't call with the `-s` flag:

```
  def restartcmd
    if Puppet::Util::Package.versioncmp(Facter.value(:kernelrelease), '5.11') >= 0
      [command(:adm), :restart, "-s", @resource[:name]]
    else
      # Solaris 10 does not have synchronous restart
      [command(:adm), :restart, @resource[:name]]
    end
  end
```

To fix it, you can just add the following to line 179:

```
Facter.stubs(:value).with(:kernelrelease).returns '5.11'
```

Additionally, you can remove the `Facter.instance_variable` and `Facter.add` bits and use the same method as above to set the `kernelrelease` fact.